### PR TITLE
Backport Calico related fixes to release 1.11.

### LINKED ISSUE #65045: Fail to run Calico 3.0+ with existing addon yaml files under cluster/addons/calico-policy-controller/
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:

The K8s cluster with NETWORK_POLICY_PROVIDER=calico could not be brought up in GCE when using Calico 3.0+ release. It would complain a number of issues, e.g., missing CRDs, authentication issues, ...

**What you expected to happen**:

K8s cluster can be brought up with Calico 3.0+ release

**How to reproduce it (as minimally and precisely as possible)**:

- Change image versions in cluster/addons/calico-policy-controller/{calico-node-daemonset, typha-depolyment}.yaml to Calico 3.0+ release

- make quick-release

- set ENV vars

export KUBE_CLUSTER_INITIALIZATION_TIMEOUT=3600
export NUM_NODES=1
export KUBE_UP_AUTOMATIC_CLEANUP=true
export KUBE_GCE_ZONE=us-west1-a
export KUBE_GCE_ENABLE_IP_ALIASES=true
export NETWORK_POLICY_PROVIDER=calico

- run cluster/kube-up.sh to spin up a K8s cluster on GCE

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration: gce
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by jingax10**
**What this PR does / why we need it**:
We need to run Calico 3.0+ on GCP and we need to customize MTU when Calico CNI is used on GCP.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #65045 #65067

**Special notes for your reviewer**:

**Release note**:

```release-note
"NONE"
```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -2312,7 +2329,7 @@ function start-kube-addons {
     if [ -n "${CUSTOM_KUBE_PROXY_YAML:-}" ]; then
       # Replace with custom GKE kube proxy.
       cat > "$src_dir/kube-proxy/kube-proxy-ds.yaml" <<EOF
-$(echo "$CUSTOM_KUBE_PROXY_YAML")
+$CUSTOM_KUBE_PROXY_YAML
```
Comment (bowei) [inline cluster/gce/gci/configure-helper.sh]: this looks to be unrelated to calico

Comment (jingax10) [reply cluster/gce/gci/configure-helper.sh]: Thanks for pointing it out! Reverted. PTAL

**Comment by foxish** [comment]
Please note that this will miss 1.11.1 unless it's approved and in by tomorrow EOD.

**Comment by krzyzacy** [comment]
/unassign
/assign @foxish 
@foxish should give the cherrypick approve label

**Comment by jingax10** [comment]
@foxish Could you please take a look? I really need to merge it to address a couple issues on GKE. Thanks!

**Comment by foxish** [comment]
Done, thanks!

**Comment by jingax10** [comment]
kind/bug
priority/important-soon
